Source/relay/TourGuide/Items of the Month/2017/GenieBottle.ash

```diff
@@ -1,20 +1,32 @@
 RegisterResourceGenerationFunction("IOTMGenieBottleGenerateResource");
 void IOTMGenieBottleGenerateResource(ChecklistEntry [int] resource_entries)
 {
-    if (lookupItem("genie bottle").item_amount() + lookupItem("pocket wish").item_amount() == 0) return;
+    if (lookupItem("genie bottle").item_amount() + lookupItem("pocket wish").item_amount() + lookupItem("replica genie bottle").item_amount() == 0) return;
     
     int wishes_left = 0;
     if (__misc_state["in run"] && in_ronin())
         wishes_left += lookupItem("pocket wish").item_amount();
     if (lookupItem("genie bottle").item_amount() > 0 && mafiaIsPastRevision(18219) && my_path().id != PATH_BEES_HATE_YOU)
         wishes_left += clampi(3 - get_property_int("_genieWishesUsed"), 0, 3);
+    if (lookupItem("replica genie bottle").item_amount() > 0)
+        wishes_left += clampi(3 - get_property_int("_genieWishesUsed"), 0, 3);
+
     string [int] description;
     
     if (wishes_left > 0)
     {
+        // Start with URL to the classic genie bottle
         string url = "inv_use.php?pwd=" + my_hash() + "&whichitem=9529";
-        if (lookupItem("genie bottle").item_amount() == 0 || get_property_int("_genieWishesUsed") >= 3)
+
+        // Change URL if they have a replica available
+        if (lookupItem("replica genie bottle").item_amount() > 0) {
+            url = "inv_use.php?pwd="+my_hash()+"&whichitem=11234";
```

Comment:
Optional nit: would be nice to do the same thing here you're doing for garbage tote, where the item ID is the only thing that's different across classic/replica/pocket wish

Reply:
will do this tonight, this feels like a reasonable shift. (i had not written the tote tile when i did this, because i did these in year order for some reason, so that is part of why some of the later tiles feel "better" at handling the various stupid replica things than the earlier ones)

Reply:
resolved in new version